esmvaltool/utils/cmorizers/obs/cmorize_obs.py

```diff
@@ -0,0 +1,262 @@
+"""
+Run the CMORization module as a utility executable.
+
+This utility allows the user to call and execute CMOR reformatting
+scripts (support for NCL and Python at the moment), that will use
+two I/O variables passed by this utility: an input directory as
+specified in config-user.yml by the RAWOBS key, and an output dir
+created in the form of output_dir/CMOR_DATE_TIME/TierTIER/DATASET.
+The user can specify a list of DATASETS that the CMOR reformatting
+can by run on by using -o (--obs-list-cmorize) command line argument.
+The CMOR reformatting scripts are to be found in:
+esmvaltool/cmor/cmorizers/obs
+"""
+import argparse
+import logging
+import os
+import sys
+import datetime
+import subprocess
+
+from esmvaltool._task import write_ncl_settings
+from esmvaltool._config import read_config_user_file
+
+logger = logging.getLogger(__name__)
+
+HEADER = r"""
+______________________________________________________________________
+          _____ ____  __  ____     __    _ _____           _
+         | ____/ ___||  \/  \ \   / /_ _| |_   _|__   ___ | |
+         |  _| \___ \| |\/| |\ \ / / _` | | | |/ _ \ / _ \| |
+         | |___ ___) | |  | | \ V / (_| | | | | (_) | (_) | |
+         |_____|____/|_|  |_|  \_/ \__,_|_| |_|\___/ \___/|_|
+______________________________________________________________________
+
+""" + __doc__
+
+
+def _assemble_datasets(raw_obs, obs_list):
+    """Get my datasets as dictionary keyed on Tier."""
+    # check for desired datasets only (if any)
+    # if not, walk all over rawobs dir
+    # assume a RAWOBS/TierX/DATASET input structure
+    tiers = []
+    datasets = {}
+
+    # get all available tiers in source dir
+    for _, tier, _ in os.walk(raw_obs, followlinks=True):
+        tiers.append(tier)
+
+    # if user specified obs list
+    if obs_list:
+        for tier in tiers[0]:
+            datasets[tier] = []
+            for dataset_name in obs_list.split(','):
+                if os.path.isdir(os.path.join(raw_obs, tier, dataset_name)):
+                    datasets[tier].append(dataset_name)
+
+    # otherwise go through the whole raw_obs dir
+    else:
+        for tier in tiers[0]:
+            datasets[tier] = []
+            for _, dats, _ in os.walk(os.path.join(raw_obs, tier),
```

Comment:
This will not work correctly with unexpected directory structures. Use `os.listdir(os.path.join(raw_obs, tier)` to list the available datasets instead.